src/components/include/sl.h

```diff
@@ -321,6 +328,31 @@ sl_timeout_wakeup_expired(cycles_t now)
 	} while (heap_size(sl_timeout_heap()));
 }
 
+static inline int
+sl_thd_activate(struct sl_thd *t, sched_tok_t tok)
+{
+	struct cos_defcompinfo *dci = cos_defcompinfo_curr_get();
+	struct cos_compinfo    *ci  = &dci->ci;
+
+	switch(t->type) {
+	case SL_THD_THD:
+	case SL_THD_AEP:
+	{
+		return cos_defswitch(sl_thd_thdcap(t), t->prio, sl__globals()->timeout_next, tok);	
+	}
+	case SL_THD_AEP_TCAP:
+	{
+		return cos_switch(sl_thd_thdcap(t), sl_thd_tcap(t), t->prio, sl__globals()->timeout_next, sl__globals()->sched_rcv, tok);	
```

Comment:
Changed types to flags: `FLAG_SEND` to asnd instead of switch and `FLAG_OWNTC` to use a dedicated tcap for this thread.